test/_linux.py

```diff
@@ -472,6 +480,73 @@ def test_procfs_path(self):
             psutil.PROCFS_PATH = "/proc"
             os.rmdir(tdir)
 
+    def test_psutil_is_reloadable(self):
+        imp.reload(psutil)
+
+    def test_no_procfs_for_import(self):
+        my_procfs = tempfile.mkdtemp()
+
+        with open(os.path.join(my_procfs, 'stat'), 'w') as f:
+            f.write('cpu   0 0 0 0 0 0 0 0 0 0\n')
+            f.write('cpu0  0 0 0 0 0 0 0 0 0 0\n')
+            f.write('cpu1  0 0 0 0 0 0 0 0 0 0\n')
+
+        self.assertNotAlmostEqual(psutil.cpu_percent(), 0)
+        self.assertNotAlmostEqual(sum(psutil.cpu_times_percent()), 0)
+        try:
+            orig_open = open
+
+            def open_mock(name, *args):
+                if name.startswith('/proc'):
+                    # simulate an ENOENT
+                    raise IOError('rejecting access to /proc')
+                return orig_open(name, *args)
+            patch_point = 'builtins.open' if PY3 else '__builtin__.open'
+            with mock.patch(patch_point, side_effect=open_mock):
+                imp.reload(psutil)
+
+                self.assertRaises(IOError, psutil.cpu_times)
+                self.assertRaises(IOError, psutil.cpu_times, percpu=True)
+                self.assertRaises(IOError, psutil.cpu_percent)
+                self.assertRaises(IOError, psutil.cpu_percent, percpu=True)
+                self.assertRaises(IOError, psutil.cpu_times_percent)
+                self.assertRaises(
+                    IOError, psutil.cpu_times_percent, percpu=True)
+
+                psutil.PROCFS_PATH = my_procfs
+
+                self.assertAlmostEqual(psutil.cpu_percent(), 0)
+                self.assertAlmostEqual(sum(psutil.cpu_times_percent()), 0)
```

Comment:
Here I'm testing that the cpu_percent is returning a reasonable default when it was unable to load /proc/stat at import-time.

I suppose it should be safe to check that it's specifically the floating point number 0, but as there is arithmetic going on under the hood I'm not sure we want the test to fail if the immediate post-import call happens to return a very small non-zero number such as `0.00000001`.

If you want to assert that the test always returns precisely 0.0f instead, I'm happy to make that change.